apps/go-mod-validator/src/index.ts

```diff
@@ -0,0 +1,55 @@
+import * as fs from "fs";
+import { getSmartcontractkitDependencies } from "./parser";
+import { validateDependency } from "./github";
+
+async function run() {
+  // Access command-line arguments
+  const args = process.argv.slice(2);
+  if (args.length == 0) {
+    console.error("need 1 argument as the input");
+    process.exit(1);
+  }
```

Comment:
If this is to be used as a Github Action, I'd recommend looking at `@actions/core` for handling GHA inputs.

Some examples of nodejs actions:
- https://github.com/smartcontractkit/.github/blob/main/actions/signed-commits/
- https://github.com/smartcontractkit/.github/blob/main/actions/gha-workflow-validator/

Reply:
One doubt I had about this is, we won't be able to use the script outside the github workflows. In case we want to use it pre-commit check or so in future. But if that's not a problem then I will definitely implement this with the `actions.yml` in the second PR